Upstream: yes

From 95c9444f67781f83c5c22c4292a1f16559ea6337 Mon Sep 17 00:00:00 2001
From: Hannes Reinecke <hare@suse.de>
Date: Thu, 13 Mar 2014 12:50:11 +0100
Subject: [PATCH 1/1] link against libsystemd, not libsystemd-daemon

With systemd version 210 libsystemd-daemon has been merged
into libsystemd.

Signed-off-by: Hannes Reinecke <hare@suse.de>
---
 libmultipath/Makefile | 6 +++++-
 multipathd/Makefile   | 6 +++++-
 2 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/libmultipath/Makefile b/libmultipath/Makefile
index 6364364..4b4f707 100644
--- a/libmultipath/Makefile
+++ b/libmultipath/Makefile
@@ -9,7 +9,11 @@ DEVLIB = libmultipath.so
 LIBS = $(DEVLIB).$(SONAME)
 LIBDEPS = -lpthread -ldl -ldevmapper -ludev
 ifdef SYSTEMD
-	LIBDEPS += -lsystemd-daemon
+	ifeq ($(shell test $(SYSTEMD) -gt 209 && echo 1), 1)
+		LIBDEPS += -lsystemd
+	else
+		LIBDEPS += -lsystemd-daemon
+	endif
 endif
 
 OBJS = memory.o parser.o vector.o devmapper.o callout.o \
diff --git a/multipathd/Makefile b/multipathd/Makefile
index 781122a..901d1e4 100644
--- a/multipathd/Makefile
+++ b/multipathd/Makefile
@@ -11,7 +11,11 @@ ifdef SYSTEMD
 endif
 LDFLAGS += -lpthread -ldevmapper -lreadline
 ifdef SYSTEMD
-	LDFLAGS += -lsystemd-daemon
+	ifeq ($(shell test $(SYSTEMD) -gt 209 && echo 1), 1)
+		LDFLAGS += -lsystemd
+	else
+		LDFLAGS += -lsystemd-daemon
+	endif
 endif
 LDFLAGS += -ludev -ldl \
 	-L$(multipathdir) -lmultipath -L$(mpathpersistdir) -lmpathpersist
-- 
2.1.4

